allennlp/data/tokenizers/pretrained_transformer_tokenizer.py

```diff
@@ -37,20 +37,25 @@ class PretrainedTransformerTokenizer(Tokenizer):
     def __init__(
         self,
         model_name: str,
-        do_lowercase: bool,
+        do_lowercase: bool = None,
         start_tokens: List[str] = None,
         end_tokens: List[str] = None,
     ) -> None:
-        if model_name.endswith("-cased") and do_lowercase:
-            logger.warning(
-                "Your pretrained model appears to be cased, "
-                "but your tokenizer is lowercasing tokens."
-            )
-        elif model_name.endswith("-uncased") and not do_lowercase:
-            logger.warning(
-                "Your pretrained model appears to be uncased, "
-                "but your tokenizer is not lowercasing tokens."
-            )
+        name_do_lowercase = "-uncased" in model_name
```

Comment:
I think `guessed_do_lowercase` is a little bit more readable here.  "name" is too ambiguous to be immediately obvious what this variable name means.

Also, can you move this into a separate function?  It'll be more self-contained and easier to improve later that way.  Something like `do_lowercase = _get_casing(model_name, do_lowercase)`.  And in that function, you can have something like `if do_lowercase is None: return guessed_do_lowercase`, to simplify the if/else blocks a bit.

Reply:
I'm, currently working on it. I added some tests, because that do_lowercase parameter does not matter at all, since tokenizer ignores it, if it doesn't match model casing.